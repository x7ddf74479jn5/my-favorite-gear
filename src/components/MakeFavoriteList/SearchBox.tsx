import SearchIcon from "@mui/icons-material/Search";
import Container from "@mui/material/Container";
import IconButton from "@mui/material/IconButton";
import InputBase from "@mui/material/InputBase";
import Paper from "@mui/material/Paper";
import type { VFC } from "react";
import React, { memo, useState } from "react";

interface SearchBoxProps {
  handler: (str: string) => void;
}
const SearchBox: VFC<SearchBoxProps> = ({ handler }) => {
  const [term, setTerm] = useState<string>("");
  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    handler(term);
  };
  return (
    <Container maxWidth="xs">
      <Paper
        component="form"
        sx={{ display: "flex", margin: (theme) => theme.spacing(1) }}
        onSubmit={onSubmit}
      >
        <InputBase
          fullWidth
          sx={{ flex: 1, paddingLeft: (theme) => theme.spacing(1) }}
          placeholder="商品を検索"
          inputProps={{ "aria-label": "商品を検索" }}
          value={term}
          onChange={(e) => {
            setTerm(e.target.value);
          }}
        />
        <IconButton type="submit" aria-label="search">
          <SearchIcon />
        </IconButton>
      </Paper>
    </Container>
  );
};

export default memo(SearchBox);
