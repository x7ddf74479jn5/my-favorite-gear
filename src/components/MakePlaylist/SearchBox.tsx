import Container from "@material-ui/core/Container";
import IconButton from "@material-ui/core/IconButton";
import InputBase from "@material-ui/core/InputBase";
import Paper from "@material-ui/core/Paper";
import type { Theme } from "@material-ui/core/styles";
import { createStyles, makeStyles } from "@material-ui/core/styles";
import SearchIcon from "@material-ui/icons/Search";
import type { FC } from "react";
import React, { useState } from "react";

const useStyles = makeStyles((theme: Theme) => {
  return createStyles({
    root: { display: "flex", margin: theme.spacing(1) },
    input: { flex: 1, paddingLeft: theme.spacing(1) },
  });
});
interface SearchBoxProps {
  handler: (str: string) => void;
}
const SearchBox: FC<SearchBoxProps> = ({ handler }) => {
  const classes = useStyles();
  const [term, setTerm] = useState<string>("");
  const onSubmit = (event: React.FormEvent<HTMLDivElement>) => {
    event.preventDefault();
    handler(term);
  };
  return (
    <Container maxWidth="xs">
      <Paper component="form" className={classes.root} onSubmit={onSubmit}>
        <InputBase
          fullWidth
          className={classes.input}
          placeholder="商品を検索"
          inputProps={{ "aria-label": "商品を検索" }}
          value={term}
          onChange={(e) => {
            setTerm(e.target.value);
          }}
        />
        <IconButton type="submit" aria-label="search">
          <SearchIcon />
        </IconButton>
      </Paper>
    </Container>
  );
};

export default SearchBox;
