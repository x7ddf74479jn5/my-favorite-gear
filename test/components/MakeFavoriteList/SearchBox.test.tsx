import userEvent from "@testing-library/user-event";
import React from "react";
import { describe, expect, it, vi } from "vitest";

import SearchBox from "@/components/MakeFavoriteList/SearchBox";

import { render } from "../../test-utils";

const mockHandler = vi.fn();

describe("SearchBox", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<SearchBox handler={mockHandler} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { container, getByRole } = render(
      <SearchBox handler={mockHandler} />
    );

    expect(container.getElementsByTagName("form")[0]).toHaveAttribute(
      "class",
      expect.stringContaining("root")
    );

    const inputElement = getByRole("textbox", {
      name: "商品を検索",
    });

    expect(inputElement).toHaveAttribute("aria-label", "商品を検索");
    expect(inputElement).toHaveAttribute("placeholder", "商品を検索");
    expect(inputElement).toHaveAttribute("type", "text");
    expect(inputElement).toHaveValue("");

    const buttonElement = getByRole("button");
    expect(buttonElement).toHaveAttribute("aria-label", "search");
    expect(buttonElement).toHaveAttribute("type", "submit");
    expect(buttonElement).toBeEnabled();

    expect(container.getElementsByTagName("svg")).toBeTruthy();
  });

  it("triggers a submit event", async () => {
    const user = userEvent.setup();

    const { getByRole } = render(<SearchBox handler={mockHandler} />);

    const inputElement = getByRole("textbox", {
      name: "商品を検索",
    });

    await user.type(inputElement, "test");
    expect(inputElement).toHaveValue("test");
    const buttonElement = getByRole("button");
    await user.click(buttonElement);
    expect(mockHandler).toHaveBeenCalledTimes(1);
    expect(mockHandler).toBeCalledWith("test");
  });
});
