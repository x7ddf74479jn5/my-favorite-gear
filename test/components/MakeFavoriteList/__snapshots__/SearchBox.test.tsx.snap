// Vitest Snapshot v1

exports[`SearchBox > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  >
    <form
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-wdnrrz-MuiPaper-root"
    >
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-x8r86-MuiInputBase-root"
      >
        <input
          aria-label="商品を検索"
          class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
          placeholder="商品を検索"
          type="text"
          value=""
        />
      </div>
      <button
        aria-label="search"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="submit"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</DocumentFragment>
`;
